File: my-streams-application/src/main/java/nl/elstarit/demo/streams/domain/Todo.java
package nl.elstarit.demo.streams.domain;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.redis.core.RedisHash;

import javax.validation.constraints.NotBlank;
import java.io.Serializable;
import java.time.LocalDateTime;

@Document
@Data
@Builder
public class Todo implements Serializable {

  @Id
  private String id;

  @NotBlank
  private String subject;

  @NotBlank
  private String description;

  private LocalDateTime created;

  private boolean completed;
}

File: nodejs-streams/server/domain/todo.ts
export class Todo {
  private _id: string;
  private subject: string;
  private description:string;
  private created: Date;
  private completed: boolean;
  private _class: string;

  constructor(id: string, subject: string, description: string) {
    if(id  != null) {
      this._id = id;
    }
    this.subject = subject;
    this.description = description;

    this._class = "nl.elstarit.demo.streams.domain.Todo";
  }

  public setCreated(date: Date){
    this.created = date;
  }

  public setCompleted(completed : boolean){
    this.completed = completed;
  }

}
